src/main.zig

```diff
@@ -1092,6 +1219,116 @@ fn Generator(comptime WriterType: type) type {
             }
         }
 
+        fn generateMethodParams(self: *Self, method: Method) !void {
+            var first = true;
+            if (method.instance) {
+                try self.writer.print("self_: *T", .{});
+                first = false;
+            }
+            for (method.params.items) |param| {
+                if (!first)
+                    try self.writer.writeAll(", ");
+                first = false;
+                try self.generateMethodParam(param);
+            }
+        }
+
+        fn generateMethodParam(self: *Self, param: Param) !void {
+            if (getBlockType(param)) |f| {
+                try self.writer.writeAll("context: anytype, comptime ");
+                try self.writer.print("{s}_: ", .{param.name});
+                try self.writer.writeAll("fn (ctx: @TypeOf(context)");
+                for (f.params.items) |param_ty| {
+                    try self.writer.writeAll(", _: ");
+                    try self.generateType(param_ty);
+                }
+                try self.writer.writeAll(") ");
+                try self.generateType(f.return_type.*);
+            } else {
+                try self.writer.print("{s}_: ", .{param.name});
+                try self.generateType(param.ty);
+            }
+        }
+
+        fn generateBlockHelpers(self: *Self, method: Method) !void {
+            for (method.params.items) |param| {
+                if (getBlockType(param)) |f| {
+                    try self.writer.writeAll("                const Literal = ns.BlockLiteral(@TypeOf(context));\n");
+                    try self.writer.writeAll("                const Helper = struct {\n");
+                    try self.writer.writeAll("                    pub fn cCallback(literal: *Literal");
+                    for (f.params.items, 0..) |param_ty, i| {
+                        try self.writer.print(", a{d}: ", .{i});
+                        try self.generateType(param_ty);
+                    }
+                    try self.writer.writeAll(") callconv(.C) void {\n");
+                    try self.writer.print("                        {s}_(literal.context", .{param.name});
+                    for (0..f.params.items.len) |i| {
+                        try self.writer.print(", a{d}", .{i});
+                    }
+                    try self.writer.writeAll(");\n");
+                    try self.writer.writeAll("                    }\n");
+                    try self.writer.writeAll("                };\n");
+                    try self.writer.writeAll("                const descriptor = ns.BlockDescriptor{ .reserved = 0, .size = @sizeOf(Literal) };\n");
```

Comment:
Out of curiosity, are you running `zig fmt` on the output? It'd be good to make sure this matches up

Reply:
I do run `zig fmt` before checking in - I'm pretty sure this lines up reasonably well though.